### PR TITLE
Correct and improve TS type definitions

### PR DESCRIPTION
**Pull request by iota-pi**
Corrected, updated and improved TypeScript type definition file

### CODE REVIEW

File: dist/pptxgen.d.ts [hunk 1]
```diff
@@ -34,67 +34,325 @@ declare class PptxGenJS {
...
-  export interface DataOrPath {
-    // Exactly one must be set
-    data?: string;
-    path?: string;
-  }
-  export interface ImageOptions extends CommonOptions, DataOrPath {
+
+  export type DataOrPath = { data: string; path?: undefined } | { data?: undefined; path: string };
+
+  export interface ImageBaseOptions extends CommonOptions {
```
Comment (adrienarcuri) [inline dist/pptxgen.d.ts]: Hello @iota-pi 
Thank you for improving TS definition.

Could you insert Hyperlink and ImageSizingOptions in your Pull Request please ?
`
  export interface ImageBaseOptions extends CommonOptions {
    hyperlink?: Hyperlink;
    rounding?: boolean;
    sizing?: ImageSizingOptions;
  }

  export interface ImageSizingOptions extends CommonOptions {
    type?: 'cover' | 'contain' | 'crop';
  }

  export type Hyperlink = ({ url: string; slide?: undefined } | { slide: number; url?: undefined }) & { tooltip?: string };
`

File: dist/pptxgen.d.ts [hunk 2]
```diff
@@ -34,67 +34,325 @@ declare class PptxGenJS {
...
+    fontSize?: number;
+  }
+
+  export interface TextOptions extends ShapeOptions, BasicTextFormatting {
+    align?: HAlign;
+    autoFit?: boolean;
+    breakLine?: boolean;
+    bullet?: boolean | { type: 'number'; code?: undefined } | { code: string; type?: undefined };
+    charSpacing?: number;
+    hyperlink?: ({ url: string; slide?: undefined } | { slide: number; url?: undefined }) & { tooltip?: string };
```
Comment (adrienarcuri) [inline dist/pptxgen.d.ts]: We can use Hyperlink type:
` hyperlink?: Hyperlink;`

**Comment by iota-pi** [comment]
Also renamed `CommonOptions` to `PositionOptions` for clarity, but added alias for backwards compatibility

**Comment by gitbrent** [comment]
Thanks @iota-pi and @adrienarcuri !

**Comment by thitemple** [comment]
Hey there, quick question, do you thing it would be helpful to make the shape names a string literal? Something like this:

```
export type ShapeType = "LINE" | "RECTANGLE" | "OVAL" | ...;
```

I'd be glad to add that

**Comment by iota-pi** [comment]
There are rather a lot of them... ideally the `pptxgen.shapes.js` file would be a JSON or TS file and we could just do
```
export type ShapeType = keyof gObjPptxShapes;
```

**Comment by thitemple** [comment]
We can't do that today because that file is a JS file. Turning it into a JSON or TS file would add even more changes and I'm not sure if that's the author wants if that's the case I could work on that.

If not, I have that literal type defined already, I could just add it to the typings file.

**Comment by gitbrent** [comment]
Actually, the author has rewritten the library in TS already (the 3.0 branch) and would appreciate any help with definitions, etc. :-)

**Pull request merged**